exchanges/exchange.go

```diff
@@ -1659,3 +1660,18 @@ func (b *Base) Shutdown() error {
 	}
 	return b.Requester.Shutdown()
 }
+
+// MatchSymbolWithAvailablePairs returns a currency pair based on the supplied
+// symbol and asset type. If the string is expected to have a delimiter this
+// will attempt to screen it out.
+func (b *Base) MatchSymbolWithAvailablePairs(symbol string, a asset.Item, hasDelimiter bool) (currency.Pair, error) {
```

Comment:
So utilising the same benchmarking as last time for comparison:
```go
func BenchmarkNewPairFromString(b *testing.B) {
	b.ReportAllocs()
	for i := 0; i < b.N; i++ {
		_, _ = currency.NewPairFromString("ZTG_USDT")
	}
}

func BenchmarkDeriveFrom(b *testing.B) {
	b.ReportAllocs()
	for i := 0; i < b.N; i++ {
		avail, _ := g.GetAvailablePairs(asset.Spot)
		_, _ = avail.DeriveFrom(strings.Replace("ZTG_USDT", "_", "", 1))
	}
}

func BenchmarkDeriveFromCheating(b *testing.B) {
	b.ReportAllocs()
	avail, _ := g.GetAvailablePairs(asset.Spot)
	for i := 0; i < b.N; i++ {
		_, _ = avail.DeriveFrom(strings.Replace("ZTG_USDT", "_", "", 1))
	}
}

func BenchmarkMatchSymbolWithAvailablePairs(b *testing.B) {
	b.ReportAllocs()
	for i := 0; i < b.N; i++ {
		_, _ = g.MatchSymbolWithAvailablePairs("ZTG_USDT", asset.Spot, true)
	}
}
```

| Benchmark | times tested | time taken | Bytes per op | Allocations |
|-------------|--------------|-------------|--------------|-------------|
|BenchmarkNewPairFromString-10|20982969|57.07 ns/op|4 B/op| 1 allocs/op|
|BenchmarkDeriveFrom-10|17596|67605 ns/op|294928 B/op|4 allocs/op|
|BenchmarkDeriveFromCheating-10 |156580|7163 ns/op|17 B/op|2 allocs/op|
|BenchmarkMatchSymbolWithAvailablePairs-10|15378223|76.50 ns/op|16 B/op|2 allocs/op|


Does show a lovely improvement. Would you be considering the removal of `Derivefrom`? Because your new feature seems underutilised in this PR